core/authentication_api.php

```diff
@@ -717,6 +717,20 @@ function auth_does_password_match( $p_user_id, $p_test_password ) {
 	}
 
 	$t_password = user_get_field( $p_user_id, 'password' );
+
+	# Process SAFE_HASH separately from the legacy methods, to safeguard against
+	# timing attacks leveraging password_verify() function.
+	if(    SAFE_HASH == $t_configured_login_method
+		&& password_verify( $p_test_password, $t_password )
+	) {
+		# Update the password hash if it does not match the current algorithm
+		if( password_needs_rehash( $t_password, PASSWORD_DEFAULT ) ) {
+			user_set_password( $p_user_id, $p_test_password, true );
```

Comment:
This is not new.  But the `user_set_password` assumes that it is being called by a user and hence it does change the cookie which is not necessary since we are just no hashing.  So no reason to log user out of other browsers for example.  Technically speaking, we don't need to worry about verification token, but that should have no side effects.